src/components/ChallengePage/ChallengePage.jsx

```diff
@@ -0,0 +1,65 @@
+import React, { useState } from "react";
+import { mockChallenges } from "../../mock-data/mockChallenges";
+import ChallengeCard from "../ChallengeCard/ChallengeCard";
+import DashboardNavbar from "../../components/DashboardNavbar/DashboardNavbar";
+import filterIcon from "../../assets/images/filterIcon.svg";
+import "./ChallengePage.scss";
+
+const ChallengePage = () => {
+  const [selectedCategories, setSelectedCategories] = useState([
+    "Development",
+    "Python",
+    "React",
+    "Java",
+    "DSA",
+  ]);
+  const [selectedDifficulties, setSelectedDifficulties] = useState([
+    "Easy",
+    "Intermediate",
+    "Hard",
+  ]);
+
+  // Filter challenges based on selected categories and difficulties
+  const filteredChallenges = mockChallenges.challenges.filter(
+    (challenge) =>
+      selectedCategories.includes(challenge.category) &&
+      selectedDifficulties.includes(challenge.difficulty)
+  );
+
+  return (
+    <>
+      <DashboardNavbar />
+      <div className="challenge-page">
+        <div className="header">
+          <h1>Challenges</h1>
+          <div className="filter">
+            <img src={filterIcon} alt="Filter Icon" />
+            <span>Filter</span>
+          </div>
+        </div>
+
+        {/* Display selected filters as tabs */}
+        <div className="selected-filters">
+          {selectedCategories.map((category) => (
+            <span key={category} className="filter-tab">
+              {category}
+            </span>
+          ))}
+          {selectedDifficulties.map((difficulty) => (
+            <span key={difficulty} className="filter-tab">
+              {difficulty}
+            </span>
+          ))}
+        </div>
+
+        <div className="card-container">
+          {filteredChallenges.map((challenge, index) => (
+            <ChallengeCard key={index} challenge={challenge} />
```

Comment:
Might not be best practice to use the index as a key but find a unique identifier or something else instead.

Reply:
if there is an unique ID then use it. Maybe even combine the challenge name with the index to make it unique. 

Index is ok if it's some list that doesn't change much in an isolated case. Like some small component.